iceoryx_posh/test/moduletests/test_popo_toml_gateway_config_parser.cpp

```diff
@@ -255,3 +332,88 @@ TEST_P(TomlGatewayConfigParserTest, ParseMalformedInputFileCausesError)
+
+    auto serviceEntry = cpptoml::make_table();
+    serviceEntry->insert("service", "service");
+    serviceEntry->insert("instance", "instance");
+    serviceEntry->insert("event", "event");
+
+    auto serviceEntry1 = cpptoml::make_table();
+    serviceEntry1->insert("service", "service");
+    serviceEntry1->insert("instance", "instance");
+    serviceEntry1->insert("event", "event");
+    serviceArray->push_back(serviceEntry1);
+
+    toml->insert("services", serviceArray);
+    CreateTmpTomlFile(toml);
+
+    iox::roudi::ConfigFilePathString_t Path =
+    iox::roudi::ConfigFilePathString_t(iox::cxx::TruncateToCapacity, TestFilePath);
+    auto result = TomlGatewayConfigParser::parse(Path);
+    GatewayConfig config = result.value();
+    EXPECT_FALSE(result.has_error());
+    EXPECT_FALSE(config.m_configuredServices.empty());
+    EXPECT_EQ(config.m_configuredServices.size(), 1);
+}
+
+TEST_F(TomlGatewayConfigParserSuiteTest, ParseValidConfigFileWithMaximumAllowedNumberOfConfiguredServicesReturnNoError)
+{
+    auto toml = cpptoml::make_table();
+    auto serviceArray = cpptoml::make_table_array();
+    auto serviceEntry = cpptoml::make_table();
+
+    for (uint32_t i = 1U; i <= iox::MAX_GATEWAY_SERVICES; ++i)
+    {
+        std::string stringentry = "validservice"+ std::to_string(i);
+        serviceEntry->insert("service", stringentry);
+        serviceEntry->insert("instance", stringentry);
+        serviceEntry->insert("event", stringentry);
+        serviceArray->push_back(serviceEntry);
+    }
+
+    toml->insert("services", serviceArray);
+    CreateTmpTomlFile(toml);
+
+    iox::roudi::ConfigFilePathString_t Path =
+    iox::roudi::ConfigFilePathString_t(iox::cxx::TruncateToCapacity, TestFilePath);
+    auto result = TomlGatewayConfigParser::parse(Path);
+    GatewayConfig config = result.value();
+
+    EXPECT_EQ(config.m_configuredServices.size(), iox::MAX_GATEWAY_SERVICES);
+    EXPECT_FALSE(result.has_error());
+    EXPECT_FALSE(config.m_configuredServices.empty());
+
+}
+
+TEST_F(TomlGatewayConfigParserSuiteTest, ParseValidConfigFileWithMoreThanMaximumAllowedNumberOfConfiguredServicesReturnOnlyMaximumAllowedNumber)
+{
+    auto toml = cpptoml::make_table();
+    auto serviceArray = cpptoml::make_table_array();
+    auto serviceEntry = cpptoml::make_table();
+
+    for (uint32_t i = 1U; i <= (iox::MAX_GATEWAY_SERVICES + 2); ++i)
+    {
+        std::string stringentry = "validservice"+ std::to_string(i);
+        serviceEntry->insert("service", stringentry);
+        serviceEntry->insert("instance", stringentry);
+        serviceEntry->insert("event", stringentry);
+        serviceArray->push_back(serviceEntry);
+    }
+
+    toml->insert("services", serviceArray);
+    CreateTmpTomlFile(toml);
+
+    iox::roudi::ConfigFilePathString_t Path =
+    iox::roudi::ConfigFilePathString_t(iox::cxx::TruncateToCapacity, TestFilePath);
+    auto result = TomlGatewayConfigParser::parse(Path);
```

Comment:
In this particular case, I would add the entry `MAXIMUM_NUMBER_OF_ENTRIES_EXCEEDED` into `TomlGatewayConfigParseError` enum and return it here. Otherwise, the user has no idea that two services are missing.

Reply:
@elfenpiff  what do you mean by return it here, you would like to check the exceeded number of entries in parse function no
?

Reply:
@manel94 I meant that `TomlGatewayConfigParserError` returns `MAXIMUM_NUMBER_OF_ENTRIES_EXCEEDED` when there are more entries in the config file present then the parser can handle.